File: modules/turbo-deviceinfo/android/src/main/java/com/turbodeviceinfo/DeviceInfoCore.kt
package com.turbodeviceinfo

import android.content.Context
import android.os.Build
import android.util.Log

/**
 * DeviceInfoCore - Pure native business logic (minimal dependencies)
 * 
 * This class contains all device info logic.
 * It can be called from:
 * - TurboDeviceInfoModule (TurboModule for JavaScript)
 * - ModuleInterop (for other native modules like cg-webview)
 * - Unit tests
 */
class DeviceInfoCore private constructor(private val context: Context) {
    
    init {
        Log.d(TAG, "🚀 [DeviceInfoCore] Initialized (Pure native, minimal RN dependencies)")
    }
    
    // ========================================
    // BUSINESS LOGIC
    // ========================================
    
    fun getDeviceName(): String {
        return Build.MODEL
    }
    
    fun getSystemVersion(): String {
        return Build.VERSION.RELEASE
    }
    
    fun getBundleId(): String {
        return context.packageName
    }
    
    fun getDeviceModel(): String {
        val model = "${Build.MANUFACTURER} ${Build.MODEL}"
        Log.d(TAG, "🔵 [DeviceInfoCore] getDeviceModel() = $model")
        return model
    }
    
    companion object {
        private const val TAG = "DeviceInfoCore"
        
        @Volatile
        private var instance: DeviceInfoCore? = null
        
        @JvmStatic
        fun getInstance(context: Context): DeviceInfoCore {
            return instance ?: synchronized(this) {
                instance ?: DeviceInfoCore(context.applicationContext).also { instance = it }
            }
        }
    }
}

File: modules/turbo-deviceinfo/ios/DeviceInfoCore.m
#import "DeviceInfoCore.h"
#import <UIKit/UIKit.h>

@implementation DeviceInfoCore

+ (instancetype)shared {
    static DeviceInfoCore *sharedInstance = nil;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        sharedInstance = [[self alloc] init];
    });
    return sharedInstance;
}

- (instancetype)init {
    if (self = [super init]) {
        NSLog(@"🚀 [DeviceInfoCore] Initialized (Pure native, no RN dependencies)");
    }
    return self;
}

// ========================================
// BUSINESS LOGIC
// ========================================

- (NSString *)getDeviceName {
    return [[UIDevice currentDevice] name];
}

- (NSString *)getSystemVersion {
    return [[UIDevice currentDevice] systemVersion];
}

- (NSString *)getBundleId {
    return [[NSBundle mainBundle] bundleIdentifier];
}

- (NSString *)getDeviceModel {
    NSString *model = [[UIDevice currentDevice] model];
    NSLog(@"🔵 [DeviceInfoCore] getDeviceModel() = %@", model);
    return model;
}

@end

